### PR TITLE
Jaylen ho luc unit test

### PR DESCRIPTION
**Pull request by JaylenHoLuc**

### CODE REVIEW

File: packages/instant-apps-components/src/components/instant-apps-create/instant-apps-create.test.ts [hunk 1]
```diff
@@ -10,18 +11,35 @@ const scene = new WebScene({
     id: '0614ea1f9dd043e9ba157b9c20d3c538',
   },
 });
+const testCreateOptions = test.extend<{
+  create: HTMLInstantAppsCreateElement;
+  shadow: ShadowRoot;
+}>({
+  create: async ({}, use) => {
+    const element = document.createElement('instant-apps-create');
+    element.setAttribute('show-scale-bar', 'true');
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-create/instant-apps-create.test.ts:20]: Does `show-scale-bar` need to be set here?

File: packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts [hunk 2]
```diff
@@ -1,88 +1,196 @@
...
+  type allTests = cases[];
 
-    expect(element?.iconImageScale).toBe('s');
-    expect(element?.iconImage).toBe('https://www.esri.com/content/dam/esrisites/en-us/arcgis/products/arcgis-instant-apps/assets/instant-apps-banner-fg.png');
-  });
+  const testCases: allTests = [
+    [
+      false,
+      function createNoSignIn() {
+        let ss = document.createElement('instant-apps-landing-page');
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts:45]: What does `ss` stand for?

File: packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts [hunk 3]
```diff
@@ -1,88 +1,196 @@
...
+        return ss;
+      },
+    ],
+    [
+      true,
+      function createSignIn() {
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts:51]: `createNoSignIn` and `createSignIn` looks like it can be one function with an arg that determines whether to enable/disable sign in.

File: packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts [hunk 4]
```diff
@@ -1,88 +1,196 @@
...
-  });
-  test('port and oauthappid', async () => {
-    landingPage.oauthappid = 'wfblKnj2mvxLOiCx';
-    const element = document.querySelector('instant-apps-landing-page');
-    expect(element).toBeTruthy();
-    expect(element?.oauthappid).toBe(landingPage.oauthappid);
+          expect(element?.open).toBe(false);
+        });
+      });
+      describe.runIf(!elem)('sign in disabled subtitleText', async () => {
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-landing-page/instant-apps-landing-page.test.ts:110]: 🆒

File: packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts [hunk 5]
```diff
@@ -1,319 +1,253 @@
...
+  const { center, level, viewpoint, selectedFeature, hiddenLayers } = shareParams;
+
+  type cases = [string, () => HTMLInstantAppsSocialShareElement];
+  type allTests = cases[];
+
+  const testCases: allTests = [
+    [
+      'popover',
+      function createPopover() {
+        let ss = document.createElement('instant-apps-social-share');
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts:23]: What does `ss` stand for? Can you make this const name more descriptive to what it is?

File: packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts [hunk 6]
```diff
@@ -1,319 +1,253 @@
...
+  };
+  const { center, level, viewpoint, selectedFeature, hiddenLayers } = shareParams;
+
+  type cases = [string, () => HTMLInstantAppsSocialShareElement];
+  type allTests = cases[];
+
+  const testCases: allTests = [
+    [
+      'popover',
+      function createPopover() {
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts:22]: createPopover/createInline can probably be one function

File: packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts [hunk 7]
```diff
@@ -1,319 +1,253 @@
...
+          expect(popover).toBeTruthy();
+        });
+        test('shareButtonColor', async () => {
+          expect(element).toBeTruthy();
+          expect(element?.shareButtonColor).toBe('neutral');
+          socialShare.shareButtonColor = 'inverse';
+          document.body.append(socialShare);
+          await new Promise(resolve => requestIdleCallback(resolve));
+          expect(element?.shareButtonColor).toBe('inverse');
+          const kind = shadow.querySelector("calcite-button[id='shareButton']").getAttribute('kind');
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts:81]: why not use `getElementById`?

File: packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts [hunk 8]
```diff
@@ -1,319 +1,253 @@
...
+          expect(kind).toBe(socialShare.shareButtonColor);
+        });
+
+        test('shareButtonType', async () => {
+          expect(element).toBeTruthy();
+          expect(element?.shareButtonType).toBe('button');
+          const button = shadow.querySelector("calcite-button[id='shareButton']");
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts:88]: why not use getElementById?

File: packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts [hunk 9]
```diff
@@ -1,319 +1,253 @@
...
+          expect(button).toBeTruthy();
+
+          socialShare.shareButtonType = 'action';
+          document.body.append(socialShare);
+          await new Promise(resolve => requestIdleCallback(resolve));
+          const action = shadow.querySelector("calcite-action[id='shareButton']");
```
Comment (rslibed) [inline packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts:94]: why not use getElementById?

File: packages/instant-apps-components/src/instant-apps-social-share.html [hunk 10]
```diff
@@ -57,7 +57,7 @@
         info-button="true"
         info-is-open="true"
       >
-        <instant-apps-social-share slot="actions-end" popover-button-icon-scale="s" scale="s" share-button-scale="m"></instant-apps-social-share>
+        <instant-apps-social-share mode="inline" embed="true" slot="actions-end" popover-button-icon-scale="s" scale="s" share-button-scale="m"></instant-apps-social-share>
```
Comment (rslibed) [inline packages/instant-apps-components/src/instant-apps-social-share.html:60]: can you undo this change (`mode="inline"`), this will break the header layout

**Review by rslibed — CHANGES_REQUESTED** [review]
Good start for the second round 👍

**Comment by rslibed** [comment]
Pushed updates to fix time filter test. The test webmap was reconfigured causing the test data to break.

File: packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts (hunk 9 above)
Comment (JaylenHoLuc) [reply packages/instant-apps-components/src/components/instant-apps-social-share/instant-apps-social-share.test.ts:94]: I don't think you can directly use getElementById to access elements within a shadow root

**Review by rslibed — APPROVED** [review]
✅
